SmartDeviceLinkTests/SDLSubscribeButtonManagerSpec.m

```diff
@@ -0,0 +1,716 @@
+
+                NSArray<SDLSubscribeButtonObserver *> *observers = testManager.subscribeButtonObservers[testButtonName];
+                expect(observers.count).to(equal(2));
+                expect(NSStringFromSelector(observers[0].selector)).to(equal(NSStringFromSelector(testSelector1)));
+                expect(NSStringFromSelector(observers[1].selector)).to(equal(NSStringFromSelector(testSelector2)));
+
+                expect(testConnectionManager.receivedRequests.count).to(equal(1));
+                expect([testConnectionManager.receivedRequests[0] isKindOfClass:SDLSubscribeButton.class]);
+            });
+
+            it(@"should not notify the observer when a success response is recieved for the subscribe button request", ^{
+                [testManager subscribeButton:testButtonName withObserver:testSubscribeButtonObserver1 selector:testSelector1];
+                [testConnectionManager respondToLastRequestWithResponse:testSubscribeSuccessResponse];
+
+                expect(testSubscribeButtonObserver1.selectorCalledCount).to(equal(0));
+            });
+
+            it(@"should notify the observer with the error and remove the observer when a failure response is recieved for the subscribe button request", ^{
+                [testManager subscribeButton:testButtonName withObserver:testSubscribeButtonObserver1 selector:testSelector2];
+                NSError *testError = [NSError errorWithDomain:@"errorDomain" code:9 userInfo:@{@"subscribe button error":@"error 2"}];
+                [testConnectionManager respondToLastRequestWithResponse:testSubscribeFailureResponse error:testError];
+
+                expect(testSubscribeButtonObserver1.selectorCalledCount).to(equal(1));
+                expect(testSubscribeButtonObserver1.buttonErrorsReceived.count).to(equal(1));
+                expect(testSubscribeButtonObserver1.buttonErrorsReceived[0]).to(equal(testError));
+                expect(testSubscribeButtonObserver1.buttonNamesReceived[0]).to(equal(testButtonName));
+
+                NSArray<SDLSubscribeButtonObserver *> *observers = testManager.subscribeButtonObservers[testButtonName];
+                expect(observers.count).to(equal(0));
+            });
+
+            it(@"should ignore a subscription attempt with an invalid selector - selector has too many parameters", ^{
+                SEL testInvalidSelector = @selector(buttonPressEventWithButtonName:error:buttonPress:buttonEvent:extraParameter:);
+                [testManager subscribeButton:testButtonName withObserver:testSubscribeButtonObserver1 selector:testInvalidSelector];
+
+                NSArray<SDLSubscribeButtonObserver *> *observers = testManager.subscribeButtonObservers[testButtonName];
+                expect(observers).to(beNil());
+
+                expect(testConnectionManager.receivedRequests.count).to(equal(0));
+            });
```

Comment:
Test for ignoring subscription request if observer is nil?

Reply:
Fixed. Added